### PR TITLE
feat: backfill sync customers and invoices

### PR DESCRIPTION
**Pull request by phamhieu**
- fix: handle deleted customer
- fix: sync subscriptions of all statuses
- feat: backfill sync customers and invoices

### CODE REVIEW

File: db/migrations/0015_add_customer_deleted.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+alter table stripe.customers
+    add deleted boolean default false not null;
```
Comment (joshenlim) [inline db/migrations/0015_add_customer_deleted.sql:2]: should we do a deleted_at? since we also have an updated_at column

File: src/lib/sync.ts [hunk 2]
```diff
@@ -25,10 +27,30 @@ export async function syncPrices(): Promise<{ synced: number }> {
...
-  for await (const subscription of stripe.subscriptions.list({ limit: 100 })) {
+  for await (const subscription of stripe.subscriptions.list({ status: 'all', limit: 100 })) {
     await upsertSubscription(subscription)
     synced++
   }
 
   return { synced }
 }
+
+export async function syncCustomers(): Promise<{ synced: number }> {
```
Comment (joshenlim) [inline src/lib/sync.ts:38]: just to check here: does this mean only the first 100 customers will be synced?

Comment (phamhieu) [reply src/lib/sync.ts:38]: stripe has very good [auto-pagination](https://stripe.com/docs/api/pagination/auto). That block of code will sync everything with 100 items per request

File: db/migrations/0015_add_customer_deleted.sql (hunk 1 above)
Comment (phamhieu) [reply db/migrations/0015_add_customer_deleted.sql:2]: there is no deleted_at provided. https://stripe.com/docs/api/customers/delete

File: src/lib/sync.ts (hunk 2 above)
Comment (joshenlim) [reply src/lib/sync.ts:38]: oooh wow damn never knew that

**Review by joshenlim — APPROVED** [review]

**Pull request merged**